### PR TITLE
r.horizon: add support for multiple points

### PR DESCRIPTION
**Pull request by petrasovaa**
This adds support for multiple points to make it more convenient and much faster than when rerunning r.horizon for each point (no need to read in data repetitively). To keep backwards compatibility, the CSV output looks like:

```
azimuth,horizon_height
0.000000,0.093339
20.000000,0.061608
40.000000,0.006298
...
azimuth,horizon_height
0.000000,0.004273
20.000000,0.000000
40.000000,0.000000
...
azimuth,horizon_height
0.000000,0.002732
20.000000,0.000000
40.000000,0.000000
...
```

The JSON format repeats azimuths for each point:
```
[
    {
        "x": 633760.504202,
        "y": 223703.781513,
        "azimuth": [
            0.000000,
            100.000000,
            200.000000
        ],
        "horizon_height": [
            0.093339,
            0.010001,
            0.074268
        ]
    },
    {
        "x": 634306.722689,
        "y": 222296.218487,
        "azimuth": [
            0.000000,
            100.000000,
            200.000000
        ],
        "horizon_height": [
            0.004273,
            0.003309,
            0.027359
        ]
    },
    {
        "x": 636323.529412,
        "y": 221834.033613,
        "azimuth": [
            0.000000,
            100.000000,
            200.000000
        ],
        "horizon_height": [
            0.002732,
            0.004609,
            0.032626
        ]
    }
]
```
 I am not sure if that's the best solution, but for loading into pandas this works:
```
df = pd.json_normalize(json.loads(data))
df.explode(["azimuth", "horizon_height"])
```
```
               x              y azimuth horizon_height
0  633760.504202  223703.781513     0.0       0.093339
0  633760.504202  223703.781513   100.0       0.010001
0  633760.504202  223703.781513   200.0       0.074268
1  634306.722689  222296.218487     0.0       0.004273
1  634306.722689  222296.218487   100.0       0.003309
1  634306.722689  222296.218487   200.0       0.027359
2  636323.529412  221834.033613     0.0       0.002732
2  636323.529412  221834.033613   100.0       0.004609
2  636323.529412  221834.033613   200.0       0.032626
```

### CODE REVIEW

File: raster/r.horizon/main.c [hunk 1]
```diff
@@ -575,8 +590,34 @@ int main(int argc, char *argv[])
...
+        for (int i = 0; i < point_count; ++i) {
+            /* Calculate the horizon for each point */
+            if (format == JSON) {
+                origin_value = json_value_init_object();
+                origin = json_value_get_object(origin_value);
+            }
+            calculate_point_mode(&settings, &geometry, xcoords[i], ycoords[i],
+                                 fp, format, origin);
+            if (format == JSON)
+                json_array_append_value(coordinates, origin_value);
```
[the remark this answers is not in the document]
Comment (nilason) [reply raster/r.horizon/main.c:610]: Braces are always needed with multiline code after a control statement. Braces for one-line code after an if-statement is optional, usually omitted in GRASS code. Could be clang-formatted by setting [InsertBraces](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#insertbraces) to true (it is now set to false).

[the remark this answers is not in the document]
Comment (petrasovaa) [reply raster/r.horizon/main.c:610]: Right, I was talking about single line statements. Setting it to true would probably result in a lot of changes...

[the remark this answers is not in the document]
Comment (echoix) [reply raster/r.horizon/main.c:610]: Ok, that answers it, and is pretty much what I thought it was.

**Comment by echoix** [comment]
Is anyone else able to review it, see if there's something missing? I didn't find anything, but find it ok and ready

**Review by nilason — APPROVED** [review]
Looks good to me!

**Pull request merged**